### ISSUE TITLE
Wrong place for new folder in Linux Mint 19.3 Mate

### ISSUE THREAD

**Issue by HenrikJansson1977** [mate-desktop/caja#1356]
#### Expected behaviour
I want to create a new Folder on the desktop, and I want the new folder to be placed on that spot I am holding the mouse at the moment.

#### Actual behaviour
New folders are instead placed to the left on the screen. Even if you try to create a new folder on the desktop on screen number 2 placed to the right of screen number 1, the new folder will be placed to the left on screen number 1.
This always happens with 19.2 and 19.3 Mate. With 19 and 19.1 it sometimes happens that the new folder will be placed to the left, about 1 of 10.

#### Steps to reproduce the behaviour
Right-click and create a new folder on the desktop somewhere on the screen.

#### MATE general version
henrik@henrik-HP-Compaq-8200-Elite-SFF-PC:~$ inxi -Fxd
System:
Host: henrik-HP-Compaq-8200-Elite-SFF-PC Kernel: 5.3.0-40-generic x86_64
bits: 64 compiler: gcc v: 7.4.0 Desktop: MATE 1.22.2
Distro: Linux Mint 19.3 Tricia base: Ubuntu 18.04 bionic

**Comment by lukefromdc** [comment]
I've seen similar issues with everything on the primary monitor, where if I create a new folder or file in a crowded spot it comes up somewhere else usually well to the left

**Comment by HenrikJansson1977** [comment]
What will happen now? Can someone fix this problem?

**Comment by 13rac1** [comment]
Yes, now we wait. The maintainers will prioritize it on their TODO lists. Another developer may experience the same annoyance, find this issue, and decide to fix the problem. I might be that developer, I might not. 🤔

**Comment by 13rac1** [comment]
My notes so far for anyone who wants to dig into this further. `caja_file_operations_new_folder()` is the actual function running the operation: https://github.com/mate-desktop/caja/blob/2149626bef57b8ec1bd3edea537dc4e8e7548d10/libcaja-private/caja-file-operations.c#L6274-L6279

The [GdkPoint](https://developer.gnome.org/gdk3/stable/gdk3-Points-Rectangles-and-Regions.html#GdkPoint) argument can optionally contain the correct information. https://github.com/mate-desktop/caja/blob/2149626bef57b8ec1bd3edea537dc4e8e7548d10/libcaja-private/caja-file-operations.c#L6294-L6296 Perhaps it is not being passed correctly in the Desktop view code? Or perhaps it's not set correctly? Needs further debugging.
